src/libPMacc/include/math/vector/Vector.hpp

```diff
@@ -60,6 +61,41 @@ struct Vector_components
     }
 };
 
+
+/** functor to copy a object element wise
+ *
+ * @tparam isDestConst define if destination is const (not copyable) object
+ */
+template<bool isDestConst>
+struct CopyElementWise
+{
+    /** copy object element wise
+     *
+     * @tparam T_Dest destination object type
+     * @tparam T_Src source object type
+     */
+    template<typename T_Dest,typename T_Src>
+    HDINLINE void operator()(T_Dest& dest,const T_Src& src) const
+    {
+        PMACC_CASSERT_MSG(CopyElementWise_destination_and_source_had_different_dimension,T_Dest::dim == T_Src::dim);
+        for (int d = 0; d < T_Dest::dim; d++)
+            dest[d] = src[d];
+    }
+};
+
+/** specialization for constant destination
+ *
+ * nothing is copied!
```

Comment:
that comment is a bit misleading.
one should rather write: `the constant storage is already available and set in the destination`.